test/controllers/optOut/ConfirmedOptOutControllerSpec.scala

```diff
@@ -35,12 +35,12 @@ import scala.concurrent.Future
 class ConfirmedOptOutControllerSpec extends MockAuthActions
   with MockOptOutService {
 
-  override def fakeApplication(): Application = applicationBuilderWithAuthBindings()
+  override lazy val app: Application = applicationBuilderWithAuthBindings
     .overrides(
       api.inject.bind[OptOutService].toInstance(mockOptOutService)
     ).build()
 
-  val testController = fakeApplication().injector.instanceOf[ConfirmedOptOutController]
+  val testController = app.injector.instanceOf[ConfirmedOptOutController]
```

Comment:
same thing, we are using lazy val right away

Reply:
I have updated to use lazy val but I don’t believe it was necessary as yes the “val” was called straight away but the nature of classes mean that they are only executed at run time. The issue seemed to be more around the fakeApplication() function. I expect under the covers it was being called several times and even though we was stopping the play application after every test class some files where remaining open.